metals/src/main/scala/scala/meta/internal/rename/RenameProvider.scala

```diff
@@ -101,8 +102,9 @@ final class RenameProvider(
         val newName =
           if (suggestedName.charAt(0) == '`')
             suggestedName.substring(1, suggestedName.length() - 1)
-          else suggestedName
-
+          else {
+            Identifier.backtickWrap(suggestedName)
+          }
```

Comment:
So it looks like this screws a bit with renamed imports as well. Since if you have `Java-List` already in backtics and you attempt to rename it, the `suggestedName` for it here is without ticks causing it to be wrapped in ticks and then you end up with double ticks. 

Also total nit, but can can we try to be consistent with blocks or not instead of mixing. So either

```suggestion
else Identifier.backtickWrap(suggestedName)
```
or add blocks to the if clause.

Reply:
Ok it turns out this was harder than aniticipated :sweat_smile:  I added some more tests, made the existing ones pass. Now the current behaviour will add backticks if they needed, but are not yet added.

Reply:
haha it always is.